src/server/http/setup_connection.js

```diff
@@ -54,7 +54,7 @@ export default function (kbnServer, server, config) {
       reply.redirect(formatUrl({
         port,
         protocol: 'https',
-        hostname: host,
```

Comment:
While HTTP 1.1 does say the host header is required, to be completely safe here we should probably fall back to `host` if `req.headers.host` isn't set.  So:

`hostname: req.headers.host || host,`